internal/command/user_v2_human.go

```diff
@@ -0,0 +1,540 @@
+			return err
+		}
+	}
+
+	if h.Phone != nil && h.Phone.Number != "" {
+		if h.Phone.Number, err = h.Phone.Number.Normalize(); err != nil {
+			return err
+		}
+	}
+
+	if h.Password != nil {
+		if err := h.Password.Validate(hasher); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *Password) Validate(hasher *crypto.PasswordHasher) error {
+	if p.EncodedPasswordHash != nil {
+		if !hasher.EncodingSupported(*p.EncodedPasswordHash) {
+			return errors.ThrowInvalidArgument(nil, "USER-JDk4t", "Errors.User.Password.NotSupported")
+		}
+	}
+	if p.Password == nil && p.EncodedPasswordHash == nil {
+		return errors.ThrowInvalidArgument(nil, "COMMAND-3M0fs", "Errors.User.Password.Empty")
+	}
+	if p.Password != nil && p.EncodedPasswordHash != nil {
+		return errors.ThrowInvalidArgument(nil, "COMMAND-3M0fsss", "Errors.User.Password.NotSupported")
+	}
+	return nil
+}
+
+func (c *Commands) AddUserHuman(ctx context.Context, resourceOwner string, human *AddHuman, allowInitMail bool, alg crypto.EncryptionAlgorithm) (err error) {
+	if resourceOwner == "" {
+		return errors.ThrowInvalidArgument(nil, "COMMA-5Ky74", "Errors.Internal")
+	}
+
+	if err := human.Validate(c.userPasswordHasher); err != nil {
+		return err
+	}
+
+	if human.ID == "" {
+		human.ID, err = c.idGenerator.Next()
+		if err != nil {
+			return err
+		}
+	}
+
+	existingHuman, err := c.userHumanWriteModel(
+		ctx,
+		human.ID,
+		resourceOwner,
+		true, // user profile is always necessary at creation
+		human.Email.Address != "",
+		human.Phone.Number != "",
+		human.Password != "" || human.EncodedPasswordHash != "",
+		true,  // state is always necessary at creation
+		false, // avatar not updateable
+	)
+	if err != nil {
+		return err
+	}
+	if isUserStateExists(existingHuman.UserState) {
+		return errors.ThrowPreconditionFailed(nil, "COMMAND-k2unb", "Errors.User.AlreadyExisting")
+	}
+
+	domainPolicy, err := c.domainPolicyWriteModel(ctx, resourceOwner)
+	if err != nil {
+		return err
+	}
+
+	if err = c.userValidateDomain(ctx, resourceOwner, human.Username, domainPolicy.UserLoginMustBeDomain); err != nil {
+		return err
+	}
+
+	var createCmd humanCreationCommand
+	if human.Register {
+		createCmd = user.NewHumanRegisteredEvent(
+			ctx,
+			UserAggregateFromWriteModel(&existingHuman.WriteModel),
+			human.Username,
+			human.FirstName,
+			human.LastName,
+			human.NickName,
+			human.DisplayName,
+			human.PreferredLanguage,
+			human.Gender,
+			human.Email.Address,
+			domainPolicy.UserLoginMustBeDomain,
+		)
+	} else {
+		createCmd = user.NewHumanAddedEvent(
+			ctx,
+			UserAggregateFromWriteModel(&existingHuman.WriteModel),
+			human.Username,
+			human.FirstName,
+			human.LastName,
+			human.NickName,
+			human.DisplayName,
+			human.PreferredLanguage,
+			human.Gender,
+			human.Email.Address,
+			domainPolicy.UserLoginMustBeDomain,
+		)
+	}
+
+	if human.Phone.Number != "" {
+		createCmd.AddPhoneData(human.Phone.Number)
+	}
+
+	filter := c.eventstore.Filter
```

Comment:
Use of deprecated method `Filter`. `FilterToQueryReducer` should be used instead.

Reply:
Currently left in, as if I would need to remove more code duplicated will exist, will do a followup issue which includes the cleanup from user v1